packages/realm-server/tests/billing-test.ts

```diff
@@ -0,0 +1,367 @@
+    ...addExplicitParens(separatedByCommas(valueExpressions)),
+    ` RETURNING *`,
+  ] as Expression);
+
+  return {
+    id: result[0].id as string,
+    matrixUserId: result[0].matrix_user_id as string,
+    stripeCustomerId: result[0].stripe_customer_id as string,
+  };
+}
+
+async function insertPlan(
+  dbAdapter: PgAdapter,
+  name: string,
+  monthlyPrice: number,
+  creditsIncluded: number,
+  stripePlanId: string,
+): Promise<Plan> {
+  let { valueExpressions, nameExpressions: _nameExpressions } = asExpressions({
+    name,
+    monthly_price: monthlyPrice,
+    credits_included: creditsIncluded,
+    stripe_plan_id: stripePlanId,
+  });
+  let result = await query(dbAdapter, [
+    `INSERT INTO plans (name, monthly_price, credits_included, stripe_plan_id) VALUES`,
+    ...addExplicitParens(separatedByCommas(valueExpressions)),
+    ` RETURNING *`,
+  ] as Expression);
+
+  return {
+    id: result[0].id as string,
+    name: result[0].name as string,
+    monthlyPrice: result[0].monthly_price as number,
+    creditsIncluded: result[0].credits_included as number,
+  };
+}
+
+async function fetchStripeEvents(dbAdapter: PgAdapter) {
+  return await query(dbAdapter, [`SELECT * FROM stripe_events`]);
+}
+
+async function fetchSubscriptionsByUserId(
+  dbAdapter: PgAdapter,
+  userId: string,
+): Promise<Subscription[]> {
+  let results = (await query(dbAdapter, [
+    `SELECT * FROM subscriptions WHERE user_id = `,
+    param(userId),
+  ])) as {
+    id: string;
+    user_id: string;
+    plan_id: string;
+    started_at: number;
+    ended_at: number;
+    status: string;
+    stripe_subscription_id: string;
+  }[];
+
+  return results.map((result) => ({
+    id: result.id,
+    userId: result.user_id,
+    planId: result.plan_id,
+    startedAt: result.started_at,
+    status: result.status,
+    stripeSubscriptionId: result.stripe_subscription_id,
+  }));
+}
+
+async function fetchSubscriptionCyclesBySubscriptionId(
+  dbAdapter: PgAdapter,
+  subscriptionId: string,
+): Promise<SubscriptionCycle[]> {
+  let results = await query(dbAdapter, [
+    `SELECT * FROM subscription_cycles WHERE subscription_id = `,
+    param(subscriptionId),
+  ]);
+
+  return results.map((result) => ({
+    id: result.id as string,
+    subscriptionId: result.subscription_id as string,
+    periodStart: parseInt(result.period_start as string),
+    periodEnd: parseInt(result.period_end as string),
+  }));
+}
+
+async function fetchCreditsLedgerByUser(
+  dbAdapter: PgAdapter,
+  userId: string,
+): Promise<LedgerEntry[]> {
+  let results = await query(dbAdapter, [
+    `SELECT * FROM credits_ledger WHERE user_id = `,
+    param(userId),
+  ]);
+
+  return results.map((result) => ({
+    id: result.id as string,
+    userId: result.user_id as string,
+    creditAmount: result.credit_amount as number,
+    creditType: result.credit_type as string,
+    subscriptionCycleId: result.subscription_cycle_id as string,
+  }));
+}
```

Comment:
We will probably want to extract these to a test helpers file at some point. OK to be here for now though.